filebeat/harvester/forwarder.go

```diff
@@ -21,59 +19,38 @@ type Forwarder struct {
 	Config     ForwarderConfig
 	Outlet     Outlet
 	Processors *processors.Processors
+	meta       common.MapStr
 }
 
 // ForwarderConfig contains all config options shared by all harvesters
 type ForwarderConfig struct {
-	common.EventMetadata `config:",inline"`      // Fields and tags to add to events.
-	Type                 string                  `config:"type"`
-	Pipeline             string                  `config:"pipeline"`
-	Module               string                  `config:"_module_name"`  // hidden option to set the module name
-	Fileset              string                  `config:"_fileset_name"` // hidden option to set the fileset name
-	Processors           processors.PluginConfig `config:"processors"`
+	Type string `config:"type"`
 }
 
 // NewForwarder creates a new forwarder instances and initialises processors if configured
 func NewForwarder(cfg *common.Config, outlet Outlet) (*Forwarder, error) {
-
 	config := ForwarderConfig{}
 	if err := cfg.Unpack(&config); err != nil {
 		return nil, err
 	}
 
-	processors, err := processors.New(config.Processors)
-	if err != nil {
-		return nil, err
-	}
-
 	f := &Forwarder{
-		Outlet:     outlet,
-		Processors: processors,
-		Config:     config,
+		Outlet: outlet,
+		Config: config,
 	}
 
 	return f, nil
-
 }
 
 // Send updates the prospector state and sends the event to the spooler
 // All state updates done by the prospector itself are synchronous to make sure not states are overwritten
 func (f *Forwarder) Send(data *util.Data) error {
-
 	// Add additional prospector meta data to the event
-	data.Meta.Pipeline = f.Config.Pipeline
-	data.Meta.Module = f.Config.Module
-	data.Meta.Fileset = f.Config.Fileset
-
 	if data.HasEvent() {
-		data.Event[common.EventMetadataKey] = f.Config.EventMetadata
-		data.Event.Put("prospector.type", f.Config.Type)
-
-		// run the filters before sending to spooler
-		data.Event = f.Processors.RunBC(data.Event)
+		data.Event.PutValue("prospector.type", f.Config.Type)
```

Comment:
Could we also add the type in the outlet? Probably we need the type in other places too?